In [116]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [117]:
df =pd.read_csv(r"C:\Users\USER\Desktop\Titanic\titanic.csv")
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [156]:
survival_count = df['Survived'].value_counts()
survival_count
print(survival_count[1],'passengers survived')

342 passengers survived


In [118]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

#### 1. How Many People Survived And How Many People died?


In [120]:
df['Name'].count()

891

In [121]:
df['Survived'].value_counts(0)

Survived
0    549
1    342
Name: count, dtype: int64

#### it shows 549 people died and 342 survived

#### 2. How Many Passengers were in first class, second class and third class?

In [124]:
class_count = df['Pclass'].value_counts()


print("Number of passengers in First Class:", class_count[1])
print("Number of passengers in Second Class:", class_count[2])
print("Number of passengers in Third Class:", class_count[3])


Number of passengers in First Class: 216
Number of passengers in Second Class: 184
Number of passengers in Third Class: 491


#### 3. Find the number of Male and Female passengers


In [126]:
gender_count= df['Sex'].value_counts()
print('The number of male passengers is',gender_count['male'])
print('The number of female passengers is',gender_count['female'])

The number of male passengers is 577
The number of female passengers is 314


##### 4. Who has Better Chance Of Survival?


In [128]:
## Calculate the survival rate for each gender
survival_by_gender = df.groupby('Sex')['Survived'].mean()

print("Survival rate for females:", survival_by_gender['female'])
print("Survival rate for males:", survival_by_gender['male'])


Survival rate for females: 0.7420382165605095
Survival rate for males: 0.18890814558058924


In [129]:
#### Calculate the survival rate for each pclass
survival_by_pclass = df.groupby('Pclass')['Survived'].mean()

print('Survival rate for first class:',survival_by_pclass[1])
print('Survival rate for second class:',survival_by_pclass[2])
print('Survival rate for third class:',survival_by_pclass[3])

Survival rate for first class: 0.6296296296296297
Survival rate for second class: 0.47282608695652173
Survival rate for third class: 0.24236252545824846


#### 5. Which Passenger class has better chance of Survival?

In [131]:
passenger = df.groupby('Name')['Survived'].mean().head(5)
print('The passenger with the best chance of survival is',passenger)

The passenger with the best chance of survival is Name
Abbing, Mr. Anthony                      0.0
Abbott, Mr. Rossmore Edward              0.0
Abbott, Mrs. Stanton (Rosa Hunt)         1.0
Abelson, Mr. Samuel                      0.0
Abelson, Mrs. Samuel (Hannah Wizosky)    1.0
Name: Survived, dtype: float64


In [132]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [133]:
df.pivot_table(index='Pclass',columns='Sex' ,values='Survived')

Sex       female      male
Pclass                    
1       0.968085  0.368852
2       0.921053  0.157407
3       0.500000  0.135447